helix-core/src/main/java/org/apache/helix/messaging/DefaultMessagingService.java

```diff
@@ -70,7 +70,7 @@ public DefaultMessagingService(HelixManager manager) {
 
     _taskExecutor = new HelixTaskExecutor(
         new ParticipantStatusMonitor(isParticipant, manager.getInstanceName()),
-        new MessageQueueMonitor(manager.getClusterName(), manager.getInstanceName()));
+        new MessageQueueMonitor(manager.getClusterName(), manager.getInstanceName()), manager);
```

Comment:
jiajunwang 14 hours ago Contributor
Did you miss some files here? MessageQueueMonitor does not have such a constructor and it is not changed in this PR.

 @jiajunwang
jiajunwang 14 hours ago Contributor
In general, passing the manager to the monitor object would be an overkill. If it is really necessary, then we shall change the constructor to accept the manager only, since it includes all the information.

Reply:
Resolving this conversation since we are not doing it this way now.